utils/openssl.py

```diff
@@ -0,0 +1,18 @@
+from subprocess import PIPE, Popen
+
+
+def certificate_matches_host(certificate, host):
+    common_name = get_certificate_common_name(certificate)
+    return host.endswith(common_name.replace('*.', ''))
+
+
+def get_certificate_common_name(certificate):
+    proc = Popen(
+        ['openssl', 'x509', '-noout', '-subject'],
+        stdin=PIPE,
+        stdout=PIPE,
+        stderr=PIPE
+    )
+    out, err = proc.communicate(certificate)
+
+    return out.split('/CN=')[1].strip()
```

Comment:
Somewhere in there we should probably extracts SANs and check those too. 

https://stackoverflow.com/questions/20983217/how-to-display-the-subject-alternative-name-of-a-certificate

Some certificates we (may) have would have something like `domain.com` as the CN and then a list of alternative domains in the SAN extension (ex: www.domain.com, foo.domain.com, bar.domain, ...)

Or we can wait until we hit this and refactor at that point in time

Reply:
that makes sense, and i'd prefer to solve when it hits us :)